From f5f634a4861075b1cdf21cbeee8dd6265d66b1f6 Mon Sep 17 00:00:00 2001
From: Yu-Ren Zhang <e8035669@gmail.com>
Date: Fri, 8 Sep 2023 23:23:25 +0800
Subject: [PATCH 3/3] fix platform detect

---
 libmamba/src/specs/platform.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libmamba/src/specs/platform.cpp b/libmamba/src/specs/platform.cpp
index 7ed2eb36..6b426f80 100644
--- a/libmamba/src/specs/platform.cpp
+++ b/libmamba/src/specs/platform.cpp
@@ -42,7 +42,8 @@ namespace mamba::specs
 #elif defined(__arm__) || defined(__thumb__)
 #ifdef ___ARM_ARCH_6__
         return Platform::linux_armv6l;
-#elif __ARM_ARCH_7__
+#elif defined(__ARM_ARCH_7__) || defined(__ARM_ARCH_7A__) || \
+      defined(__ARM_ARCH_7R__) || defined(__ARM_ARCH_7M__)
         return Platform::linux_armv7l;
 #else
 #error "Unknown Linux arm platform"
-- 
2.20.1

