From a2b025619c6abfd8d465d6a8a62f332a04784094 Mon Sep 17 00:00:00 2001
From: Yu-Ren Zhang <e8035669@gmail.com>
Date: Tue, 12 Sep 2023 17:45:57 +0800
Subject: [PATCH 4/5] fix get_info bug on 32 bit devices

---
 libmamba/src/core/curl.cpp | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/libmamba/src/core/curl.cpp b/libmamba/src/core/curl.cpp
index 0c837450..12809aeb 100644
--- a/libmamba/src/core/curl.cpp
+++ b/libmamba/src/core/curl.cpp
@@ -222,6 +222,7 @@ namespace mamba
     template tl::expected<long, CURLcode> CURLHandle::get_info(CURLINFO option);
     template tl::expected<char*, CURLcode> CURLHandle::get_info(CURLINFO option);
     template tl::expected<double, CURLcode> CURLHandle::get_info(CURLINFO option);
+    template tl::expected<long long, CURLcode> CURLHandle::get_info(CURLINFO option);
     template tl::expected<curl_slist*, CURLcode> CURLHandle::get_info(CURLINFO option);
 
     template <class I>
@@ -241,7 +242,15 @@ namespace mamba
     template <>
     tl::expected<std::size_t, CURLcode> CURLHandle::get_info(CURLINFO option)
     {
-        return get_integer_info<std::size_t>(option);
+        auto res = get_info<curl_off_t>(option);
+        if (res)
+        {
+            return static_cast<std::size_t>(res.value());
+        }
+        else
+        {
+            return tl::unexpected(res.error());
+        }
     }
 
     template <>
-- 
2.20.1

